File: MvpJava/src/main/java/com/droid/ui/register/RegisterActivity.java
package com.droid.ui.register;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import com.droid.R;
import com.droid.presenter.register.RegisterPresenter;
import com.droid.presenter.register.RegisterView;
import com.droid.ui.BaseMvpActivity;

public class RegisterActivity extends BaseMvpActivity<RegisterPresenter> implements RegisterView {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.register);
        mPresenter = new RegisterPresenter();
        mPresenter.mView = this;
    }

    public void click(View view) {
        mPresenter.register("zhangsan", "123456");
    }


    @Override
    public void onResult(String result) {
        System.out.println(result);
    }
}

File: MvpKotlin/src/main/java/com/droid/mvp/ui/register/RegisterActivity.kt
package com.droid.mvp.ui.register

import android.os.Bundle
import com.droid.mvp.R
import com.droid.mvp.presenter.register.RegisterPresenter
import com.droid.mvp.presenter.register.RegisterView
import com.droid.mvp.ui.BaseMvpActivity
import kotlinx.android.synthetic.main.register.*
import org.jetbrains.anko.toast

class RegisterActivity:BaseMvpActivity<RegisterPresenter>(), RegisterView {
    override fun onResult(result:String) {
        toast("Register Succeed! $result")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.register)

        mPresenter = RegisterPresenter()
        mPresenter.mView = this

        mBtn.setOnClickListener {
            mPresenter.register("zhangsan", "123456")
        }
    }
}